lib/Installation/Partitioner/SeparateHomeCheckbox.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces all accessing methods for 'Propose Separate Home
+# Partition/Volume' checkbox.
+# The checkbox is extracted to the separate package as several pages contain
+# it, but different shortcut is used for selecting.
+
+# Maintainer: Oleksandr Orlov <oorlov@suse.de>
+
+package Installation::Partitioner::SeparateHomeCheckbox;
+use strict;
+use warnings FATAL => 'all';
+use testapi;
+
+use constant {
+    CHECKED_SEPARATE_HOME_CHECKBOX   => 'enabledhome',
+    UNCHECKED_SEPARATE_HOME_CHECKBOX => 'disabledhome'
+};
+
+sub new {
+    my ($class, $shortcut) = @_;
+    my $self = bless {
+        shortcut => $shortcut || die 'Shortcut for Separate Home checkbox is not provided.'
+    }, $class;
+}
+
+sub is_checked {
+    my ($self) = @_;
+    assert_screen([(CHECKED_SEPARATE_HOME_CHECKBOX, UNCHECKED_SEPARATE_HOME_CHECKBOX)]);
+    if (match_has_tag(CHECKED_SEPARATE_HOME_CHECKBOX)) {
+        return 1;
+    }
+    elsif (match_has_tag(UNCHECKED_SEPARATE_HOME_CHECKBOX)) {
+        return 0;
+    }
+}
+
+sub set_value {
+    my ($self, $is_checked) = @_;
+    if ($is_checked eq $self->is_checked()) {
```

Comment:
you're comparing numbers I guess, perhaps `=` instead of `eq`, but normally works :)

Reply:
Good catch. I doubted here. You dispelled the doubts:)